src/plugins/workspace/public/components/global_search/search_pages_command.tsx

```diff
@@ -36,50 +51,58 @@ export const searchPages = async (
       getFirstUseCaseOfFeatureConfigs(currentWorkspace?.features || []) || ''
     );
 
-    const searchResult = allAvailableCaseId.flatMap((useCaseId) => {
-      const navGroup = navGroupMap[useCaseId];
-      if (navGroup) {
-        const links = navGroup.navLinks as Array<ChromeRegistrationNavLink & ChromeNavLink>;
-        // parent nav links are not clickable
-        const parentNavLinkIds = links.map((link) => link.parentNavLinkId).filter((link) => !!link);
-        return links
-          .filter((link) => {
-            const title = link.title;
-            let parentNavLinkTitle;
-            // parent title also taken into consideration for search its sub items
-            if (link.parentNavLinkId) {
-              parentNavLinkTitle = navGroup.navLinks.find(
-                (navLink) => navLink.id === link.parentNavLinkId
-              )?.title;
-            }
-            const titleMatch = title && title.toLowerCase().includes(query.toLowerCase());
-            const parentTitleMatch =
-              parentNavLinkTitle && parentNavLinkTitle.toLowerCase().includes(query.toLowerCase());
-            return (
-              !link.hidden &&
-              !link.disabled &&
-              (titleMatch || parentTitleMatch) &&
-              !parentNavLinkIds.includes(link.id)
-            );
-          })
-          .map((link) => ({
-            ...link,
-            navGroup,
-          }));
+    const searchResult = searchNavigationLinks(allAvailableCaseId, navGroupMap, query);
+
+    const handleCallback = (link: Link) => {
+      callback?.();
+      const isPageOutOfWorkspace = link.navGroup.type === NavGroupType.SYSTEM;
+      if (isPageOutOfWorkspace && currentWorkspace) {
+        // remove workspace information in the URL, special handling for data source which could visible both in/out workspace
+        const urlWithoutWorkspace = formatUrlWithWorkspaceId(
+          link.href,
+          '',
+          coreStart.http.basePath
+        );
+        window.location.assign(urlWithoutWorkspace);
+        return;
       }
-      return [];
-    });
+      coreStart.application.navigateToApp(link.id);
+    };
+
+    const handleBreadcrumbs = (
+      link: Link,
+      breadcrumbs: EuiBreadcrumb[],
+      availableUseCases: WorkspaceUseCase[]
+    ) => {
+      const isPageOutOfWorkspace = link.navGroup.type === NavGroupType.SYSTEM;
```

Comment:
We have a field: `workspaceAvailability` on application, and core will handle with the case where an application jump to a page outside workspace as long as you call navigateToApp / navigateToUrl method.

Reply:
Existing logic, forget it.

Reply:
This part of code was originally exists, we can leave this here for now